catroid/src/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -354,6 +354,10 @@ private Object interpretFunction(Functions function, Sprite sprite) {
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.toDegrees(Math.atan(doubleValueOfLeftChild));
 			case EXP:
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.exp(doubleValueOfLeftChild);
+			case POWER:
+				return ((doubleValueOfLeftChild == null || doubleValueOfRightChild == null)
+						|| (doubleValueOfLeftChild < 0 && doubleValueOfRightChild < 1 && doubleValueOfRightChild > -1))
```

Comment:
@robertpainsi I was able to determine the special case (NaN) we had and where the try and catch came in handy and not at the same time.
```
power(-5,(1/5))
```
will return  ```Error``` and I believe that is the Try/Catch way of handling with NaN but I dont understand why because:
```java
try {
	return java.lang.Math.pow(doubleValueOfLeftChild, doubleValueOfRightChild);
} catch (Exception e) {
	return 0d;
}
```
Should evaluate to 0 whatever exception is thrown..